internal/dbadapter/db_adapter.go

```diff
@@ -377,66 +352,53 @@ func (a *StandardAdapter) JobGetAvailableTx(ctx context.Context, tx pgx.Tx, queu
+type JobSetStateParams struct {
+	ID          int64
+	errData     []byte
+	finalizedAt *time.Time
+	maxAttempts *int
+	scheduledAt *time.Time
+	state       dbsqlc.JobState
 }
 
-func (a *StandardAdapter) JobSetCompletedIfRunning(ctx context.Context, job JobToComplete) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetCompletedIfRunning(ctx, a.executor, dbsqlc.JobSetCompletedIfRunningParams{
-		ID:          job.ID,
-		FinalizedAt: job.FinalizedAt.UTC(),
-	})
+func JobSetStateCancelled(id int64, finalizedAt time.Time, errData []byte) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, errData: errData, finalizedAt: &finalizedAt, state: dbsqlc.JobStateCancelled}
 }
 
-func (a *StandardAdapter) JobSetCompletedTx(ctx context.Context, tx pgx.Tx, id int64, completedAt time.Time) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetCompleted(ctx, tx, dbsqlc.JobSetCompletedParams{
-		ID:          id,
-		FinalizedAt: completedAt.UTC(),
-	})
+func JobSetStateCompleted(id int64, finalizedAt time.Time) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, finalizedAt: &finalizedAt, state: dbsqlc.JobStateCompleted}
 }
 
-func (a *StandardAdapter) JobSetDiscardedIfRunning(ctx context.Context, id int64, finalizedAt time.Time, err []byte) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
-
-	return a.queries.JobSetDiscardedIfRunning(ctx, a.executor, dbsqlc.JobSetDiscardedIfRunningParams{
-		ID:          id,
-		Error:       err,
-		FinalizedAt: finalizedAt,
-	})
+func JobSetStateDiscarded(id int64, finalizedAt time.Time, errData []byte) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, errData: errData, finalizedAt: &finalizedAt, state: dbsqlc.JobStateDiscarded}
 }
 
-func (a *StandardAdapter) JobSetErroredIfRunning(ctx context.Context, id int64, scheduledAt time.Time, err []byte) (*dbsqlc.RiverJob, error) {
-	ctx, cancel := context.WithTimeout(ctx, a.deadlineTimeout)
-	defer cancel()
+func JobSetStateErrored(id int64, scheduledAt time.Time, errData []byte) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, errData: errData, scheduledAt: &scheduledAt, state: dbsqlc.JobStateRetryable}
+}
 
-	return a.queries.JobSetErroredIfRunning(ctx, a.executor, dbsqlc.JobSetErroredIfRunningParams{
-		ID:          id,
-		Error:       err,
-		ScheduledAt: scheduledAt,
-	})
+func JobSetStateSnoozed(id int64, scheduledAt time.Time, maxAttempts int) *JobSetStateParams {
+	return &JobSetStateParams{ID: id, maxAttempts: &maxAttempts, scheduledAt: &scheduledAt, state: dbsqlc.JobStateScheduled}
 }
 
-func (a *StandardAdapter) JobSetSnoozedIfRunning(ctx context.Context, id int64, scheduledAt time.Time) (*dbsqlc.RiverJob, error) {
+func (a *StandardAdapter) JobSetState(ctx context.Context, params *JobSetStateParams) (*dbsqlc.RiverJob, error) {
```

Comment:
Hah, I actually had it as `*IfRunning` almost the entire time I was developing this branch, but removed it at the last second because it felt like it was adding a lot of verbosity everywhere. I don't feel strongly about it though, so I added it back in. The completer function also becomes `*IfRunning` so we're consistent everywhere.